32_lib_usage.md

```diff
@@ -3,13 +3,170 @@
 - [Utilizarea bibliotecilor externe](#utilizarea-bibliotecilor-externe)
   - [biblioteci de programare](#biblioteci-de-programare)
   - [utilizarea bibliotecilor statice](#utilizarea-bibliotecilor-statice)
+    - [Exemplu de utilizare a bibliotecii statice](#exemplu-de-utilizare-a-bibliotecii-statice)
   - [utilizarea bibliotecilor dinamice](#utilizarea-bibliotecilor-dinamice)
+    - [dynamic linking](#dynamic-linking)
+    - [dynamic loading](#dynamic-loading)
   - [Bibliografie](#bibliografie)
 
 ## biblioteci de programare
 
+__Biblioteca de programare__ este un set de resurse de programare așa cum sunt funcții, clase, obiecte, constante și variabile. În special, constantele pot fi date grafice sau multimedia. Bibliotecile de programare sunt împărțite în trei tipuri: __biblioteci statice__, __biblioteci dinamice__ și __biblioteci de șabloane__.
+
+Biblioteci statice sunt prezentate prin două fișiere:
+
+1. Fișierul antet cu extensia `h` (`hxx`, `hpp`), care descrie interfața bibliotecii;
+2. Arhiva binară cu extensia `lib` (în Windows) sau `a` (în sistemele compatibile Unix), care stochează toate resursele de programare sub formă de cod obiect.
+
+Biblioteci dinamice sunt prezentate prin trei fișiere:
+
+1. Fișierul antet cu extensia `h` (`hxx`, `hpp`), care descrie interfața bibliotecii;
+2. Arhiva binară cu extensia `dll` (în Windows) sau `so` (în sistemele compatibile Unix), numită bibliotecă dinamică (în sistemele Unix - partajată), care stochează toate resursele de programare sub formă de cod binar;
+3. Fișierul cu extensia `lib` (în Windows) sau `a` (în sistemele compatibile Unix), numit bibliotecă de import, care stochează locațiile resurselor de programare în biblioteca dinamică. În sistemele Unix, biblioteca de import nu este utilizată.
+
+Există și biblioteci care nu pot fi reprezentate altfel decât sub formă de cod sursă - acestea sunt bibliotecile care conțin șabloane de clase și funcții. Aceste biblioteci sunt furnizate în fișiere antet.
+
+Numirea bibliotecilor de programare pentru compilatorul Gnu C++ (g++) are următoarea formă:

```

Comment:
```suggestion
Numirea bibliotecilor de programare pentru compilatorul GCC pentru C++ (g++) are următoarea formă:
```

>GCC, the GNU Compiler Collection

Reply:
Gnu C++ compiler (compilator Gnu C++) este denumirea compilatorului, corectarea nu poate fi acceptata.

Reply:
> [Most of the compilers for languages other than C have their own names. The C++ compiler is G++](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/G_002b_002b-and-GCC.html)

> [When we talk about compiling one of those languages, we might refer to that compiler by its own name, or as GCC. Either is correct.](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/G_002b_002b-and-GCC.html)

Se spune corect `G++` sau simplu `GCC`. `Gnu C++ compiler` nu este un termen oficial.

GNU se scrie cu majuscule, NU cu minuscule.

> [Currently the main GCC distribution contains front ends for C (gcc), C++ (g++), Objective C, Fortran, Ada (GNAT), Go, and D.](https://gcc.gnu.org/frontends.html)

`G++` nu este un compilator singular, ci o componentă (front-end) a colecției de compilatoare GCC. Deci pot să zic `GCC pentru C++ (g++)`.

> Corectarea nu poate fi acceptata

Aștept mai multe argumentări.

Reply:
Chiar in prima referinta si este raspuns: `GCC stands for “GNU Compiler Collection”`. Deci nu poti sa zici GNU Compiler Collection for C++, fiindca GNU Compiler Collection inseamna front-end compilatoare C (punct de intrare gcc), C++ (g++), Fortran (gfortran), Ada (gnat) etc.

Se face diferenta pentru GNU C++ Compiler (g++.exe), Intel C++ Compiler (icc.exe), Microsoft C++ Compiler (cl.exe).

* `GCC` este `Gnu Compiler Collection`, tot asa se numeste si aplicatia - compilator C
* `g++` - denumirea aplicatiei - compilatorului C++